internal/cmd/testdata/scripts/executetemplate.txtar

```diff
@@ -114,6 +114,9 @@ stdout 'alphabeta'
 exec chezmoi execute-template --file $HOME${/}file.tmpl
 stdout '# contents of file.tmpl'
 
+exec chezmoi execute-template --file $HOME${/}.local/share/chezmoi/path/to/template.tmpl
+stdout '# contents of path/to/template.tmpl'
```

Comment:
Should we have a test for `.chezmoi.sourceFile` relative to `$HOME` instead of just relative to `.local/share/chezmoi`?

Reply:
I'm really curious what happens if I do `chezmoi execute-template --file /tmp/…`, but I don't think that the script execution here will do well with that, so having something for `$HOME${/}tmp/external.tmpl` with a `.chezmoi.sourceFile` inside of it would be good. We _may_ need to have that be a regex test because the `$HOME…` values are variable depending on operating system.

Also, and this is really for @twpayne, should all of the `/` in the path above be `${/}` for cross-platform testing?

Reply:
Alright, I updated the stdout patterns to be cross-platform with regard to path slashes. I checked, and `$HOME` is an artificial value created by the test harness, so it won't be different on Windows: https://github.com/twpayne/chezmoi/blob/b46fe4e5d87cc868867eb1bef5c59b0c994aa7c7/internal/cmd/main_test.go#L829-L858

I'm not sure how much the slashes actually vary in the test harness, and modern Windows accepts Unix-style slashes most places, anyways, but I modified the stdout pattern to accept either a forward or back-slash.

Reply:
@twpayne, this is ready for CI, thanks!

Reply:
> Also, and this is really for @twpayne, should all of the `/` in the path above be `${/}` for cross-platform testing?

No, it's not necessary. There are plenty of other places where this is tested.